src/PrestaShopBundle/Controller/Admin/Sell/Customer/CustomerController.php

```diff
@@ -149,7 +150,7 @@ public function createAction(Request $request)
 
                 return $this->redirectToRoute('admin_customers_index');
             }
-        } catch (CustomerException $e) {
+        } catch (Exception $e) {
```

Comment:
This has to be done in every Identifiable Object forms

Reply:
Should it catch every possible exception or only `CoreException`?

Reply:
It must be all Exceptions. @eternoendless refuses that a Module that does not handle properly exceptions breaks the page.

Reply:
I agree with the fact of protecting the exceptions from modules, however I'm afraid catching `Exception` might be a bit too wide
If another exception which has nothing to do with a module is thrown it will be caught and displayed as an alert, which can be bothersome for dev and debug (like database exception, client exception, routing, Symfony exceptions, ...)

I'm wondering instead, would it be possible to add a layer in the `HookDispatcher` so that it catch exceptions and throw a `ModuleException` wrapping the first one right away. This would allow us to catch the module exceptions independently. Quick draft:

```php
class HookDispatcher {
    public function dispatch($eventName, Event $event = null)
    {
        if ($event === null) {
            $event = new HookEvent();
        }
        if (!$event instanceof HookEvent) {
            throw new \Exception('HookDispatcher must dispatch a HookEvent subclass only. ' . get_class($event) . ' given.');
        }

        try {
             return parent::dispatch($eventName, $event);
        } catch (\Exception $e) {
             throw new ModuleException($e->getMessage(), $e->getCode(), $e);
        }
    }
}
```

Reply:
I'm not a fan 😐because:
- you oblige all exceptions coming from a hook to be ModuleException, which might make module developers think "pfff, there's no use to define different exceptions in my module, it's going to be casted to ModuleException by the core anyway"
- this PR is supposed to solve the issue we have at hands in a very simpler manner as it must be a bugfix (because it's feature freeze ❄️) and this would a massive behavioral change (=> maybe in `develop` then)

Reply:
The exception is indeed wrapped into a ModuleException but you still have access to the previous exception
But I agree it's not the best either.. I'm just concerned about catching all exceptions Then the developer (both module and core) will think "What's the point of raising an exception it will be caught as alert message anyway 😅"

Reply:
We could catch it and re-throw it in DEV, but keep it in PROD ?
Also we could log it

But this can come in `develop` 😉

Reply:
Any non-identified exception goes in the fallback error message. It should be logged in prod and probably displayed better in dev. That is an improvement that can be done in develop.

In a perfect world we'd only catch RuntimeExceptions and let LogicExceptions bubble up. But that's not the world we live in 😅...